engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationResources.scala

```diff
@@ -1,28 +1,31 @@
 package pl.touk.nussknacker.ui.security.api
 
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.directives.AuthenticationDirective
 import akka.http.scaladsl.server.{Directives, Route}
-import akka.http.scaladsl.server.directives.{AuthenticationDirective, RouteDirectives}
 import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.security.api.AuthenticationResources.LoggedUserAuth
 import sttp.client.{NothingT, SttpBackend}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-trait AuthenticationResources {
-  val directive: AuthenticationDirective[LoggedUser]
-  val route: Route = Directives.reject
-  val config: AuthenticationConfiguration
+trait AuthenticationResources extends Directives {
+  val name: String
+  val frontendSettings: ToResponseMarshallable = StatusCodes.NoContent
+
+  def authenticate(): LoggedUserAuth
+
+  final def routeWithPathPrefix(): Route = pathPrefix("authentication" / name.toLowerCase() ) { route() }
+
+  protected def route(): Route = path("settings") {
```

Comment:
let's at least add comment that settings are _really_ necessary - .i.e. one should call super.route() if overriding

Reply:
I have added a separate route named `frontendSettingsRoute` with concatenation in `routeWithPathPrefix` which is a final value.